docs/src/_reference/command-line-interface.md

```diff
@@ -1279,6 +1279,41 @@ echo '{"singer_state": {"project_123456_issues": "2020-01-01"}}' > gitlab_state.
 meltano state set --force dev:tap-gitlab-to-target-jsonl --input-file gitlab_state.json
 ```
 
+### vacuum
+
+Delete unused historical records of old runs to release some disk space. The states of jobs are not affected.
+
+#### How to use
+
+```bash
+# Delete some historical records of old runs.
+meltano state vacuum
+
+# Delete some old runs while specifying how many rows to keep per state ID.
+meltano state vacuum --rows-to-keep <row_count>
```

Comment:
```suggestion
meltano state delete --runs-to-keep <run_count>
```

I'm not sure it makes sense to have folks thinking in rows here. We currently do a [merge operation](https://docs.meltano.com/reference/command-line-interface#merge) for state, so deleting a specified number of rows would potentially result in bad state. 

If we keep it specific to runs that'd be less problematic. That said though, doing this _without_ `--pattern` could be really problematic too. 

@aaronsteers what do you think about requiring both the pattern and the run count for a destructive operation?

Reply:
I agree with using the term `runs` in place of `rows`, which is more generic.

Reply:
I totally overlooked the merge operation. I guess we need to leave all latest runs with incomplete payload and maybe plus the latest n complete states in the table, and then we want to make it clear for the user on what this command really does...